### DISCUSSION TITLE
To everybody: main branch is force pushed.

### DISCUSSION THREAD

**Discussion by drmingdrmer** [databendlabs/databend#7263] [Announcement]
Update: the second time force push updated `main` to 42de73d733c05573815c8c8ada7c3c6fa9d43226. Snippets are updated with this new one.

The original main contains several big files that have to be cleaned.
A recent history has been rewritten, and the old main branch is forcibly pushed, to clean up big objects introduced since 9a3e04561fc4648c447187493560bf92a530b519 (`tests/suites/0_stateless/13_tpch/data/*`):

Origin main:

`commit d06cc14660c82ba9bf73e95d33373467ed58c1bf (tag: v0.8.6-nightly, before-clean)`.

`main` after the first force push:

`commit 7ee0e8dfccedccadeb8f002d23035030a3d6a14b bkp-main-08-24-18-00`.

New main:

`commit 42de73d733c05573815c8c8ada7c3c6fa9d43226 (main)`.

## Update your working branch

If your current work is based on the original main, you will have to rebase your branch onto the new `main` branch(42de73d733c05573815c8c8ada7c3c6fa9d43226). E.g.:

```sh
git fetch --all
git rebase 42de73d733c05573815c8c8ada7c3c6fa9d43226 
```

**Backup your branch first!** 

## Troubleshooting

If the **rebase** mentioned above encounters a lot of conflicts, other choices are:

- If your working branch contains just a few simple commits, use cherry-pick:
  ```
  git fetch --all
  git reset --hard 42de73d733c05573815c8c8ada7c3c6fa9d43226
  git cherry-pick your-commit-1
  git cherry-pick your-commit-2
  ...
  ```

- Otherwise, try rebuilding a squashed commit upon the new `main`:

  ```
  git fetch --all
  git update-ref refs/heads/my_branch $(echo "commit_message" | git commit-tree my_branch^{tree} -p 42de73d733c05573815c8c8ada7c3c6fa9d43226)
  ```

  Replace `my_branch` and `commit_message` with your working branch and some great commit message.

**Backup first!!!**

**Comment by Xuanwo** [comment]
Or, we can close PR and start a new one (simple but works :rofl: )

**Comment by drmingdrmer** [comment]
Update: a second force push is made to clean up big objects in git history.
The main branch now is 42de73d733c05573815c8c8ada7c3c6fa9d43226.
